perun-core/src/main/java/cz/metacentrum/perun/core/impl/Activator.java

```diff
@@ -0,0 +1,82 @@
+package cz.metacentrum.perun.core.impl;
+
+import cz.metacentrum.perun.core.api.ExtSourcesManager;
+import cz.metacentrum.perun.core.api.GroupResourceAssignment;
+import cz.metacentrum.perun.core.api.GroupResourceStatus;
+import cz.metacentrum.perun.core.api.PerunClient;
+import cz.metacentrum.perun.core.api.PerunPrincipal;
+import cz.metacentrum.perun.core.api.PerunSession;
+import cz.metacentrum.perun.core.bl.PerunBl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.lang.NonNull;
+import org.springframework.scheduling.annotation.Async;
+
+import java.util.List;
+
+/**
+ * Component responsible for activating group-resource assignments in PROCESSING or
+ * FAILED state after Perun startup.
+ *
+ * @author Radoslav Čerhák <r.cerhak@gmail.com>
+ */
+public class Activator implements ApplicationListener<ContextRefreshedEvent> {
```

Comment:
The `Activator` name could be a bit confusing since it doesn't have an additional context about what is being activated.  You can consider renaming it, maybe to something like `ResourceAssignmentActivator` ? If you do so, also please update the commit message.

Reply:
Renamed to ResourceAssignmentActivator, commit message was also updated.